yummy_recipes/views.py

```diff
@@ -80,48 +86,154 @@ def categories():
         if categ_form.validate_on_submit():
             category_name = categ_form.category_name.data
 
-            if category_name in USERS[creator].categories:
+            if category_name in user_categs.values():
                 flash('That category already exists')            
             else:
                 new_category = RecipeCategory(category_name)
-                user_categs[new_category] = new_category
-
+                user_categs[new_category.category_id] = new_category
+                
             return redirect(url_for('categories'))
 
     else:
         is_auth = False
-        return redirect(url_for('signup'))     
-    
-    return render_template('recipes.html', is_auth=is_auth, categ_form=categ_form, 
-                            categs=user_categs)
+        
+    return render_template('recipes.html', is_auth=is_auth, 
+                            categ_form=categ_form, categs=user_categs)
 
 @APP.route('/recipes', methods=['POST', 'GET'])
 def recipes():
     ''' Individual recipes '''
-    form = RecipeForm()
-
+    rec_form = RecipeForm()
```

Comment:
Name your variables such that another person can understand what your variables are used for.